### DISCUSSION TITLE
Bypassing Cloudflare with a VPN

### DISCUSSION THREAD

**Discussion by vbgitrepos** [seleniumbase/SeleniumBase#3026] [Q&A] [answered]
Hi, I'm using the code below.
It works perfectly without using ExpressVPN (https://www.expressvpn.com/).
But when I turn on the VPN, it cannot bypass Cloudflare's "Verify you are human" check anymore.
Is it possible to bypass that with the VPN on?

```python
from seleniumbase import SB

with SB(uc=True, ad_block_on=True) as sb:
    url = "URL_WITH_A_CLOUDFLARE_SECURITY_CHECK"
    sb.uc_open_with_reconnect(url, 8)
```

**Comment by mdmintz** [comment, marked as answer]
To bypass Cloudflare when using a VPN (such as ExpressVPN), Add `sb.uc_gui_click_captcha()` after calling `sb.uc_open_with_reconnect(url, 8)`.

The full code from your example would look something like this:

```python
from seleniumbase import SB

with SB(uc=True, ad_block_on=True) as sb:
    url = "URL_WITH_A_CLOUDFLARE_SECURITY_CHECK"
    sb.uc_open_with_reconnect(url, 8)
    sb.uc_gui_click_captcha()
```

Adjust the `reconnect_time`, (the `8`), as needed. Most sites won't need higher than `4` seconds.

**Reply by mdmintz** [reply]
[Here's the link to the original discussion](https://web.archive.org/web/20240827194910/https://github.com/seleniumbase/SeleniumBase/discussions/3026).
